pack/handler.go

```diff
@@ -32,11 +32,13 @@ func PostPack(w http.ResponseWriter, r *http.Request) {
 
 	defer r.Body.Close()
 	pack := &Pack{}
-	if err := json.NewDecoder(r.Body).Decode(pack); err != nil {
+
+	if err := decode(r, pack); err != nil {
```

Comment:
We are combining two tasks in here one of them is to umarshal json and another is to validate user input. We should keep these separate.